### PR TITLE
build: add new version mechanism based on VERSION file and a version.sh script

### PR DESCRIPTION
**Pull request by obnoxxx**
If this is a git checkout, and git is available, then git describe is
used. Otherwise, of a VERSION.dist file exists, it is used. As a last
resort, the new checked in VERSION file is taken for the version.

This way, we can add a VERSION.dist file when creating a tarball (make
dist) without needing to overwrite the VERSION file.

This mechanism uses a version.sh script loosely inspired by
http://git.musl-libc.org/cgit/musl/tree/tools/version.sh

### CODE REVIEW

**Comment by obnoxxx** [comment]
updated without the VERSION.dist for now ...

File: scripts/version.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+BINDIR=$(dirname $(readlink -f "$0"))
```
[the remark this answers is not in the document]
Comment (obnoxxx) [reply scripts/version.sh]: fixed

**Comment by rofl0r** [comment]
LGTM :+1:

**Comment by rofl0r** [comment]
it seems annoyingly that the version variable in m4 is being cached. can we force the version.sh evaluation to be made on every run ?

**Pull request merged**